src/Uno.UI.Composition/Composition/Visual.skia.cs

```diff
@@ -178,7 +195,17 @@ internal virtual void Render(in DrawingSession parentSession)
 		}
 
 		using var session = BeginDrawing(in parentSession);
-		Draw(in session);
+		if (_requiresRepaint)
+		{
+			_requiresRepaint = false;
+			_recorder ??= new SKPictureRecorder();
+			var recorderSession = session with { Canvas = _recorder.BeginRecording(new SKRect(float.NegativeInfinity, float.NegativeInfinity, float.PositiveInfinity, float.PositiveInfinity)) };
+			// To debug what exactly gets repainted, replace the following line with `Draw(in session);`
+			Draw(in recorderSession);
+			_picture = _recorder.EndRecording();
```

Comment:
Should we consider to have a try/finally for this?

Reply:
I don't think so. External code cannot access the internal painting logic, so an exception here means that something internal went catastrophically wrong, so I suppose we should actually crash.